diff --git a/eval/fobos_sdr_recorder_main.c b/eval/fobos_sdr_recorder_main.c
index 34f1f41..9004cb5 100644
--- a/eval/fobos_sdr_recorder_main.c
+++ b/eval/fobos_sdr_recorder_main.c
@@ -45,7 +45,7 @@ void test_recorder(void)
 {
     struct fobos_sdr_dev_t* dev = NULL;
     int result = 0;
-    char lib_version[32];
+    char lib_version[64];
     char drv_version[32];
     char serials[256] = {0};
 
